File: mobile/src/main/java/treehou/se/habit/module/HasActivitySubcomponentBuilders.java
package treehou.se.habit.module;

import android.app.Activity;
import android.support.v4.app.Fragment;

public interface HasActivitySubcomponentBuilders {
    ActivityComponentBuilder getActivityComponentBuilder(Class<? extends Activity> activityClass);
    FragmentComponentBuilder getFragmentComponentBuilder(Class<? extends Fragment> fragmentClass);
}

File: mobile/src/main/java/treehou/se/habit/dagger/HasActivitySubcomponentBuilders.kt
package treehou.se.habit.dagger

import android.app.Activity
import android.support.v4.app.Fragment

interface HasActivitySubcomponentBuilders {
    fun getActivityComponentBuilder(activityClass: Class<out Activity>): ActivityComponentBuilder<*, *>
    fun getFragmentComponentBuilder(fragmentClass: Class<out Fragment>): FragmentComponentBuilder<*, *>
}